### PR TITLE
Added new chains tuning options to the setup script

### PR DESCRIPTION
**Pull request by siddardh-ra**
Added new Chains options were added to the setup script. This will aid us in testing the performance impact on Chains performance.

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/lgtm

File: ci-scripts/setup-cluster.sh [hunk 1]
```diff
@@ -123,6 +127,21 @@ EOF
...
+            chains_perf_options+="\"--kube-api-qps=$chains_kube_api_qps\","
+        fi
+        if [ -n "$chains_kube_api_burst" ]; then
+            chains_perf_options+="\"--kube-api-burst=$chains_kube_api_burst\","
+        fi
+        if [ -n "$chains_threads_per_controller" ]; then
+            chains_perf_options+="\"--threads-per-controller=$chains_threads_per_controller\""
+        fi
+    fi
+    kubectl patch TektonConfig/config --type merge --patch '{"spec":{"chain":{"options":{"deployments":{"tekton-chains-controller":{"spec":{"template":{"spec":{"containers":[{"name":"tekton-chains-controller","args":['$chains_perf_options']}]}}}}}}}}}'
```
Comment (jhutar) [inline ci-scripts/setup-cluster.sh:143]: If `$chains_perf_options` would be empty, there is no reason in running this `kubectl patch ...`.

File: ci-scripts/setup-cluster.sh [hunk 2]
```diff
@@ -123,6 +127,21 @@ EOF
...
+    chains_perf_options=""
+    if [ -n "$chains_kube_api_qps" ] || [ -n "$chains_kube_api_burst" ] || [ -n "$chains_threads_per_controller" ]; then
+        if [ -n "$chains_kube_api_qps" ]; then
+            chains_perf_options+="\"--kube-api-qps=$chains_kube_api_qps\","
+        fi
+        if [ -n "$chains_kube_api_burst" ]; then
+            chains_perf_options+="\"--kube-api-burst=$chains_kube_api_burst\","
+        fi
+        if [ -n "$chains_threads_per_controller" ]; then
+            chains_perf_options+="\"--threads-per-controller=$chains_threads_per_controller\""
```
Comment (jhutar) [inline ci-scripts/setup-cluster.sh:140]: If `$chains_threads_per_controller` would not be set, you would get coma at the end of the `$chains_perf_options` which would be invalid JSON.

File: ci-scripts/setup-cluster.sh [hunk 3]
```diff
@@ -123,6 +127,21 @@ EOF
     wait_for_entity_by_selector 300 openshift-pipelines pod app=tekton-pipelines-webhook
     kubectl -n openshift-pipelines wait --for=condition=ready --timeout=300s pod -l app=tekton-pipelines-webhook
 
+    info "Enable performance options"
+    chains_perf_options=""
+    if [ -n "$chains_kube_api_qps" ] || [ -n "$chains_kube_api_burst" ] || [ -n "$chains_threads_per_controller" ]; then
```
Comment (jhutar) [inline ci-scripts/setup-cluster.sh:132]: IMO this is extra `if` which is not needed as you are checking individual variables below in individual ifs anyway.

**Review by jhutar — APPROVED** [review]
Looks great! Added some comments, but they are not critical. I do not want to block merging because of them, so choose yourself if you want to work on these or merge it - I do not know what is your time plan.

**Pull request merged**